api/api-doc/storage_service.json

```diff
@@ -746,6 +746,54 @@
             }
          ]
       },
+      {
+          "path":"/storage_service/backup",
+          "operations":[
+              {
+                  "method":"POST",
+                  "summary":"Starts copying SSTables from a specified keyspace to a designated bucket in object storage",
+                  "type":"string",
+                  "nickname":"start_backup",
+                  "produces":[
+                      "application/json"
+                  ],
+                  "parameters":[
+                      {
+                          "name":"endpoint",
+                          "description":"ID of the configured object storage endpoint to copy sstables to",
+                          "required":true,
+                          "allowMultiple":false,
+                          "type":"string",
+                          "paramType":"query"
+                      },
```

Comment:
We should have a parameter "type" since multiple object storage protocols exist.

However, we can add it later as an optional parameter and default it to S3, since the majority are compatible with S3.

Reply:
Type is implicitly encoded in `endpoint` one. It points to an entry in object-store.yaml config which should declare what kind of endpoint it is

Reply:
that was also my impression. but what `endpoint` has is but the "endpoint" as documented at https://docs.aws.amazon.com/general/latest/gr/s3.html . it does not have a schema / protocol prefix , like "s3://s3.us-east-2.amazonaws.com". please note the protocol is not HTTP or HTTPS here. what a typical `object-storage.yaml` looks like is:
```yaml
      endpoints:
        - name: s3.us-east-2.amazonaws.com
          port: 443
          https: true
          aws_region: us-east-2
          aws_access_key_id: xxxxx
          aws_secret_access_key: xxxxx
          aws_session_token: xxxxx
```
by looking at it, one cannot tell which object store backend a given endpoint is backed. so, probably we can add a "type" field in the "endpoint" instead? and continue using "name" for the endpoint. other backends could use a different identifier or convention.

Reply:
If we're going to support object-storages with different types, then this "type" should become part of this config, not the API endpoint. In the end, this config should list endpoints by abstract "labels" and the exact URL to connect to it should become yet another parameter (see https://github.com/scylladb/scylladb/issues/13022)